File: tests/benchmarks/resolution/fixtures/fsharp/Repository.fs
module Repository

open System.Collections.Generic

type UserRecord = { Name: string; Email: string; Age: int }

let private store = Dictionary<string, UserRecord>()

let save (uid: string) (record: UserRecord) =
    store.[uid] <- record

let findById (uid: string) =
    match store.TryGetValue(uid) with
    | true, record -> Some record
    | _ -> None

let delete (uid: string) =
    store.Remove(uid) |> ignore

let count () =
    store.Count

File: tests/benchmarks/resolution/fixtures/ocaml/repository.ml
type user_record = {
  name : string;
  email : string;
  age : int;
}

let store : (string, user_record) Hashtbl.t = Hashtbl.create 16

let save uid record =
  Hashtbl.replace store uid record

let find_by_id uid =
  Hashtbl.find_opt store uid

let delete uid =
  Hashtbl.remove store uid

let count () =
  Hashtbl.length store
